lib/src/mt_mcast.c

```diff
@@ -489,3 +731,24 @@ int mt_mcast_l2_leave(struct mtl_main_impl* impl, struct rte_ether_addr* addr,
   if (mt_drv_use_kernel_ctl(impl, port)) return 0;
   return mcast_inf_remove_mac(inf, addr);
 }
+
+int mt_mcast_parse(struct mtl_main_impl* impl, struct mcast_mb_query_v3* query,
+                   enum mtl_port port) {
+  if (query->type != MEMBERSHIP_QUERY) {
+    err("%s, invalid type %u, only allow igmp query packet.\n", __func__, query->type);
+  }
+
+  uint16_t checksum = mcast_msg_checksum(MEMBERSHIP_QUERY, query, 0);
+  if (checksum != ntohs(query->checksum)) {
+    err("%s, err checksum %d:%d\n", __func__, ntohs(query->checksum), checksum);
+    return -EIO;
+  }
+
+  info("%s, received igmp query\n", __func__);
```

Comment:
Use `info_once`? Not know if it will output many logs.

Reply:
sure